### ISSUE TITLE
Inconsistency between CSI chart versions and app readme

### ISSUE THREAD

**Issue by rishabhmsra** [rancher/rancher#36027]
**Rancher Server Setup**
- Rancher version: `v2.5.12-rc1`
- Installation option (Docker install/Helm Chart): Docker

**Information about the Cluster**
- Kubernetes version: `v1.18.20-rancher1-3`
- Cluster Type (Local/Downstream): `Downstream vSphere node driver cluster with 3 worker, 1 etcd, 1 cp`
 
**Describe the bug**
On downstream cluster, CSI chart version `2.1.000` is mentioned in the readme Support Matrix but it's not available in the drop-down. (attached ss)

![csi-version](https://user-images.githubusercontent.com/36376154/148166111-5e18839b-d3c5-416a-8348-5ed2873b136f.png)

**To Reproduce**
- Create a downstream vSphere node driver cluster on k8s version 1.18.20. 
- From apps and marketplace goto CSI chart

**Comment by PennyScissors** [comment]
This is not an issue. When we did a OOB release to fix the vSphere charts versioning after Rancher 2.5.11 (see https://github.com/rancher/rancher/issues/35476), we released CSI 2.1.1 which is a copy of 2.1.000 with a `kubeVersion: <= 1.19` added, and intentionally hid 2.1.000 (see [this](https://github.com/rancher/charts/blob/a9be54baff1d1797e77d8b67bae1e28989467cf2/charts/rancher-vsphere-csi/rancher-vsphere-csi/2.1.000/Chart.yaml#L6) annotation in the chart) as 2.1.1 was meant to replace it. This is why 2.1.000 doesn't show up in the dropdown. However, at the time we did the OOB vSphere release for those charts, https://github.com/rancher/rancher/issues/34875 was causing Rancher to not respect `rancher-version` annotation constraints and all charts were showing up. So we decided to add 2.1.000 to the support matrix to make it clearer for existing users of 2.1.000. We could remove it from the support matrix since that issue has now been fixed and 2.1.000 is hidden as expected, but 2.5 is nearing EOM, and this would require updating the app-readme.md of all CSI charts and re-releasing them, which I don't think it's worth the effort.

**Comment by PennyScissors** [comment]
This is not an issue. When I did the OOB release to fix the vSphere charts versioning paths after Rancher 2.5.11 (see https://github.com/rancher/rancher/issues/35476), we released CSI 2.1.1 which is a copy of 2.1.000 with a `kubeVersion: <= 1.19` added, and intentionally hid 2.1.000 (see [this](https://github.com/rancher/charts/blob/a9be54baff1d1797e77d8b67bae1e28989467cf2/charts/rancher-vsphere-csi/rancher-vsphere-csi/2.1.000/Chart.yaml#L6) annotation in the chart) as 2.1.1 was meant to replace it. This is why 2.1.000 doesn't show up in the dropdown. However, at the time I did the OOB vSphere release for those charts, https://github.com/rancher/rancher/issues/34875 was causing Rancher to not respect `catalog.cattle.io/rancher-version` annotation constraints and all charts were showing up. So this was discussed and we decided to add 2.1.000 to the support matrix to make it clearer for existing users of 2.1.000. We could remove it from the support matrix since that issue has now been fixed and 2.1.000 is hidden as expected, but 2.5 is nearing EOM, and this would require updating the app-readme.md of all CSI charts and re-releasing them, which I don't think it's worth the effort.

**Comment by PennyScissors** [comment]
This is not an issue. When I did the OOB vSphere release to fix the vSphere charts upgrade paths after Rancher 2.5.11 (see https://github.com/rancher/rancher/issues/35476), we released CSI 2.1.1 which is a copy of 2.1.000 with the addition of a `kubeVersion: <= 1.19`, and intentionally hid 2.1.000 (see [this](https://github.com/rancher/charts/blob/a9be54baff1d1797e77d8b67bae1e28989467cf2/charts/rancher-vsphere-csi/rancher-vsphere-csi/2.1.000/Chart.yaml#L6) annotation in the chart) as 2.1.1 is meant to replace it. This is why 2.1.000 doesn't show up in the dropdown. However, at the same time this was happening, https://github.com/rancher/rancher/issues/35488 was causing Rancher to not respect `catalog.cattle.io/rancher-version` annotation constraints and all charts, including charts meant to be hidden, were showing up. So this was discussed and we decided to add 2.1.000 to the support matrix to make it clearer for existing users of 2.1.000 with clusters older than 1.19 as 35488 was going to be fixed after the vSphere release. We could remove 2.1.000 from the support matrix since 35488 issue has now been fixed and 2.1.000 is hidden as expected, but 2.5 is nearing EOM, and this would require updating the app-readme.md of all CSI charts and re-releasing them, which I don't think it's worth the effort. The recommendation is to install 2.1.1 instead.
